ext/js/language/ja/japanese.js

```diff
@@ -747,29 +747,40 @@ export function isEmphaticCodePoint(codePoint) {
  * @returns {string}
  */
 export function collapseEmphaticSequences(text, fullCollapse) {
-    let result = '';
-    let collapseCodePoint = -1;
-    for (let i = 0; i < text.length; ++i) {
-        const char = text[i];
-        const c = char.codePointAt(0) ?? -1;
-        if (isEmphaticCodePoint(c)) {
-            // Prevent match trailing emphatic
-            if (i === text.length - 1) {
-                result += char;
-                continue;
-            }
+    let left = 0;
+    while (left < text.length && isEmphaticCodePoint(/** @type {number} */ (text.codePointAt(left)))) {
```

Comment:
Don't use a typecast unless you absolutely need to; instead refactor the code to have an if undefined check with a comment saying 'text.codePointAt' cannot return 'undefined' instead